helix-term/src/commands.rs

```diff
@@ -4974,6 +4977,52 @@ pub fn extend_parent_node_start(cx: &mut Context) {
     move_node_bound_impl(cx, Direction::Backward, Movement::Extend)
 }
 
+fn select_all_impl<F>(editor: &mut Editor, select_fn: F)
+where
+    F: Fn(&Syntax, RopeSlice, Selection) -> Selection,
+{
+    let (view, doc) = current!(editor);
+
+    if let Some(syntax) = doc.syntax() {
+        let text = doc.text().slice(..);
+        let current_selection = doc.selection(view.id);
+        let selection = select_fn(syntax, text, current_selection.clone());
+        doc.set_selection(view.id, selection);
+    }
+}
+
+fn select_all_siblings(cx: &mut Context) {
+    let motion = |editor: &mut Editor| {
+        select_all_impl(editor, object::select_all_siblings);
+    };
+
+    cx.editor.apply_motion(motion);
+}
+
+fn select_all_children_in_selection(cx: &mut Context) {
+    let motion = |editor: &mut Editor| {
+        select_all_impl(editor, |syntax, text, selection| {
+            let all_children = object::select_all_children(syntax, text, selection.clone());
+
+            if selection.contains(&all_children) {
+                all_children
+            } else {
+                selection
+            }
```

Comment:
Hmm I am not sure this is quite what I had in  mind here. If I can tell this right this would just not do anything as soon as any child is not in the  selection. Instead I would like to see this be sort of like anl intersection. We should select the children covered by a selection.

In pseudo code `all_chidren.retain(|range| selection.contains(range)` (just better implemented so its not `O(n^2)`.

If this is too complex just omit this and I think I can implement this when solving #3007

Reply:
Hmm, maybe I'll just leave it out for now